### ISSUE TITLE
How to determine location of scripts/executables, and set directory where scripts/executable is run

### ISSUE THREAD

**Issue by pcarbo** [stephenslab/dsc#125]
Suppose I do

```bash
cd ~
dsc /project/mstephens/gaow/dsc-fun/benchmark.dsc
```

Where does DSC look for the scripts? From which directory are the scripts run?

For guidance it might be helpful to look at argument `knit_root_dir` in `rmarkdown::render` because it has to deal with similar issues; see `help(render)` in R.

**Comment by gaow** [comment]
First let's discuss the expected behavior:

1. working dir default: should it be current dir or the dir where one keeps DSC file? 
2. relative paths in `DSC` section: relative to working dir or to where DSC files are?

Once we are clear on this, the rest is just coding. I'll find one peaceful weekend morning to port all path related strings in DSC code to use a formal Python `path` class (a variant of `pathlib` in Python 3.6) that will resolve path related issues once and for all.

**Comment by pcarbo** [comment]
@gaow I agree it is important to come up to a consensus on this.

**Comment by gaow** [comment]
So @pcarbo you agree with my proposed behavior of 1 & 2 above?

**Comment by gaow** [comment]
Yes you are right. Okey now that we are clear about it i'll find some time to make changes to all path strings in the code.
